src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -1474,12 +1579,15 @@ protected internal virtual TypeDefinition MarkType (TypeReference reference, Dep
 						2045, sourceLocationMember, subcategory: MessageSubCategory.TrimAnalysis);
 			}
 
+			ApplyPreserveInfo (type);
```

Comment:
Why is this needed. PreserveInfo data should be static during MarkStep

Reply:
Delay loaded embedded descriptor XML can change the typepreserve.